# Assignment 10
Find your favorite news source and grab the article text.

1. Show the most common words in the article.
2. Show the most common words under a part of speech. (i.e. NOUN: {'Bob':12, 'Alice':4,})
3. Find a subject/object relationship through the dependency parser in any sentence.
4. Show the most common Entities and their types. 
5. Find Entites and their dependency (hint: entity.root.head)
6. Find the most similar words in the article

In [1]:
!pip3 install spacy

In [2]:
!pip install --upgrade spacy

Requirement already up-to-date: spacy in /Users/jMac/anaconda/lib/python3.6/site-packages
Requirement already up-to-date: requests<3.0.0,>=2.13.0 in /Users/jMac/anaconda/lib/python3.6/site-packages (from spacy)
Requirement already up-to-date: ujson>=1.35 in /Users/jMac/anaconda/lib/python3.6/site-packages (from spacy)
Requirement already up-to-date: murmurhash<0.27,>=0.26 in /Users/jMac/anaconda/lib/python3.6/site-packages (from spacy)
Requirement already up-to-date: plac<1.0.0,>=0.9.6 in /Users/jMac/anaconda/lib/python3.6/site-packages (from spacy)
Requirement already up-to-date: thinc<6.6.0,>=6.5.0 in /Users/jMac/anaconda/lib/python3.6/site-packages (from spacy)
Requirement already up-to-date: ftfy<5.0.0,>=4.4.2 in /Users/jMac/anaconda/lib/python3.6/site-packages (from spacy)
Requirement already up-to-date: dill<0.3,>=0.2 in /Users/jMac/anaconda/lib/python3.6/site-packages (from spacy)
Requirement already up-to-date: pathlib in /Users/jMac/anaconda/lib/python3.6/site-packages (from s

In [3]:
!python3 -m spacy download en




    100% |████████████████████████████████| 52.2MB 3.4MB/s ta 0:00:0111

    Linking successful

    /Users/jMac/anaconda/lib/python3.6/site-packages/en_core_web_sm/en_core_web_sm-1.2.0
    --> /Users/jMac/anaconda/lib/python3.6/site-packages/spacy/data/en

    You can now load the model via spacy.load('en').



In [4]:
import spacy

In [5]:
processor = spacy.load('en')

### After downloading the model, we can use some text from an article found on [Forbes.com](https://www.forbes.com/sites/rogergroves/2017/07/20/what-celebrity-athletes-can-learn-from-the-o-j-simpson-parole-hearing/2/#5d9647978e18). This is an article on, from a sports law standpoint,  what celebirty athletes can learn from OJ Simpson's parole hearing.

In [6]:
text2 = """Probably no other celebrity athlete will ever be in the same exact situation as O.J. Simpson – 70 years old asking for parole after a conviction for robbery, kidnapping, and conspiracy, some 23 years after one of the most sensationalized murder acquittals in American history.

But well over half of NBA and NFL players file for bankruptcy protection within 5 years of retirement. Many will have diagnosed or undiagnosed brain injuries that prevent or hamper employment, and increase depression, mood swings, and bad judgement. And bad judgment is what Simpson admitted as a cause for his botched attempt to reclaim intimate family photos and other memorabilia. Many players may face a near-perfect storm in a similar setting – a desperate need for money and a misguided attempt to retrieve it, especially since the property involves his own name, image and likeness.

Some former athletes have also lived in a cocoon of celebrity, fighting the demons of being self-absorbed. That can lead to a failure to be humble, to understand the role of contrition and remorse. Those character attributes are important when begging a judge, jury, or a parole board for forgiveness. We know crimes by former athletes defrocked from fame occur in greater number than we would like. So a version of these circumstances and lessons therefrom is not far-fetched. If convicted, instead of saying, “I’m not the bad guy, somebody else was,” the more strategic verse is “Yes, I was the bad guy, but I am not any more because…”

If athletes are paying careful attention to Simpson’s own statements, a celeb since he was 19, they should notice that he essentially attempted to be his own counsel and re-litigate the trial he lost. Instead, athletes should follow the advice of counsel to be humble, contrite, and remorseful. That is a Simpson mistake no similarly situated player should replicate.

For example, Simpson’s underlying theme throughout the hearing was that his “security” team held the guns, and perpetrated the violence, not him. Simpson repeatedly admitted the error of bad judgment in bringing thugs with him. But Simpson risked a parole board perception that he failed to take responsibility for his orchestration of the events that led to his conviction. There was a conspiracy count, so he did not have to hold a gun to be guilty. Additionally, Simpson did not testify during the robbery trial. The jury heard other testimony not so favorable. Simpson and players facing a parole hearing should remember the Board is looking at his character, and admission of the full extent of his errors. How else can one assess whether he is a risk for repeating his errors unless they appreciate the gravity and full extent of those errors.

Simpson was successful at the parole hearing. But other former stars should not be confused. Simpson had the advantage of (1) a sentence that was extraordinarily long (9.5 to 33 years for what CNN commentator Mark Geragos said is typically a one-year sentence), (2) both victims apparently being at peace with parole, (3) and incident-free actions during incarceration. Future players are not likely to have all those advantages.

Instead, athletes should focus on the strategy of humility, contrition, and remorse. More importantly, they should avoid appearing delusional about their own sordid history if one exists. Even ignoring the murder trial, Simpson still had the conviction for domestic violence, after multiple 911 calls from his then-wife. Yet Simpson’s opening statement included his self-assessment that he basically led a “conflict-free” life. He probably was thinking about the adoration of stardom he received prior to the murder trial. The delusion is that repeated violence against a woman is not a linear equation in the character analysis. It is a crime that colors the entire perception of the person, regardless of the year perpetrated.

As one board member cautioned, Simpson will receive strict scrutiny on yet-to-be-finalized conditions to his parole. I expect those conditions to have muscularity Simpson never faced on the field. He talked of being conflict-free. Starting in October, 2017, his life anew will need to be gangster-free. That too is a good lesson for former players of the future.

Simpson benefits from the law that narrows the inquiry to the crime associated with his conviction, not prior civil and criminal charges. He received a unanimous decision from the board. Under the law, that was the right decision. But he and any future athlete facing a judge, jury, or parole board, reduce their chances of success if they understate the gravity of their transgressions, especially if the crimes involve crimes against women in an era when too many athletes engage in such behavior."""

In [7]:
processed_text2 = processor(text2)

In [8]:
processed_text2

Probably no other celebrity athlete will ever be in the same exact situation as O.J. Simpson – 70 years old asking for parole after a conviction for robbery, kidnapping, and conspiracy, some 23 years after one of the most sensationalized murder acquittals in American history.

But well over half of NBA and NFL players file for bankruptcy protection within 5 years of retirement. Many will have diagnosed or undiagnosed brain injuries that prevent or hamper employment, and increase depression, mood swings, and bad judgement. And bad judgment is what Simpson admitted as a cause for his botched attempt to reclaim intimate family photos and other memorabilia. Many players may face a near-perfect storm in a similar setting – a desperate need for money and a misguided attempt to retrieve it, especially since the property involves his own name, image and likeness.

Some former athletes have also lived in a cocoon of celebrity, fighting the demons of being self-absorbed. That can lead to a failu

### 1. Find the most common words in the article

In [9]:
from spacy import attrs
from spacy.attrs import ORTH
from spacy.tokens.doc import Doc
from collections import defaultdict, Counter

**Note:** Imported various modules to help with the assignment (many found from going through the tutorials on [Spacy.io](https://spacy.io/docs/usage/resources).

In [10]:
#Store all the words that are not stop words or punctuations in a dictionary using a for loop and if statement

words = [token.text for token in processed_text2 if token.is_stop != True and token.is_punct != True]

#call on Counter module to determine word frequency

word_frequency = Counter(words)

#call and print the most common words, those top 30 words (count of the words)

common_words = word_frequency.most_common(30)
common_words


[('Simpson', 16),
 ('parole', 8),
 ('\n\n', 8),
 ('athletes', 6),
 ('players', 5),
 ('bad', 5),
 ('board', 5),
 ('years', 4),
 ('conviction', 4),
 ('trial', 4),
 ('free', 4),
 ('murder', 3),
 ('character', 3),
 ('jury', 3),
 ('crimes', 3),
 ('’s', 3),
 ('hearing', 3),
 ('violence', 3),
 ('errors', 3),
 ('celebrity', 2),
 ('athlete', 2),
 ('robbery', 2),
 ('conspiracy', 2),
 ('history', 2),
 ('judgment', 2),
 ('admitted', 2),
 ('attempt', 2),
 ('need', 2),
 ('especially', 2),
 ('self', 2)]

**Observation:** It appears from the Counter, that Simpson is the most common word with 16 mentions. Would make sense as he's the subject of the article. 

### 2. Most common words under a part of speech.

**Note:** For this portion, it's a similar approach, however we'll have to call on the parts of speech (nouns, verbs, etc.).  Let's try this for a couple parts of speech, say adjectives and nouns?

In [11]:
#Store all the words under noun and adjective parts of speech that are not stop words or punctuations in a dictionary using a for loop and if statement
#use token.pos

nouns = [token.text for token in processed_text2 if token.is_stop != True and token.is_punct != True and token.pos_ =="NOUN"]
adjectives = [token.text for token in processed_text2 if token.is_stop != True and token.is_punct != True and token.pos_ =="ADJ"]
adverbs = [token.text for token in processed_text2 if token.is_stop != True and token.is_punct != True and token.pos_ =="ADV"]

#call on Counter module to determine frequencies

noun_frequency = Counter(nouns)
adjective_frequency = Counter(adjectives)
adverb_frequency = Counter(adverbs)


In [12]:
#call and print the most common nouns, top 15 (count of the nouns)

common_nouns = noun_frequency.most_common(15)
common_nouns

[('parole', 8),
 ('athletes', 6),
 ('players', 5),
 ('board', 5),
 ('years', 4),
 ('conviction', 4),
 ('trial', 4),
 ('murder', 3),
 ('character', 3),
 ('jury', 3),
 ('crimes', 3),
 ('hearing', 3),
 ('violence', 3),
 ('errors', 3),
 ('celebrity', 2)]

In [13]:
#call and print the most common adjectives, top 15 (count of the adjectives)

common_adjectives = adjective_frequency.most_common(15)
common_adjectives

[('bad', 5),
 ('free', 4),
 ('humble', 2),
 ('exact', 1),
 ('old', 1),
 ('sensationalized', 1),
 ('American', 1),
 ('undiagnosed', 1),
 ('botched', 1),
 ('intimate', 1),
 ('perfect', 1),
 ('similar', 1),
 ('desperate', 1),
 ('misguided', 1),
 ('important', 1)]

In [14]:
#call and print the most common adverbs, top 15 (count of the adverbs)

common_adverbs = adverb_frequency.most_common(15)
common_adverbs

[('especially', 2),
 ('Instead', 2),
 ('Probably', 1),
 ('near', 1),
 ('far', 1),
 ('instead', 1),
 ('essentially', 1),
 ('similarly', 1),
 ('repeatedly', 1),
 ('Additionally', 1),
 ('extraordinarily', 1),
 ('typically', 1),
 ('apparently', 1),
 ('importantly', 1),
 ('basically', 1)]

**Observation:** Both interesting results (particularly as I have not read the article in full). From the adjective and adverb frequency counts, it would appear that this is a "lessons learned" article, warning other athlets/players from parole, conviction, trial, crimes, errors (as seen in the noun frequencies).  

### 3. Find a subject/object relationship through the dependency parser in any sentence. 

**Note:** For this portion, let's use the pr_tree against our sentences.  We can find the roots and the dependencies in our processed.text2 sentences.

In [15]:
#first, how many sentences are we looking at in our article's text?

sentences = [s for s in processed_text2.sents]
print(len(sentences)) #length of sentences

46


In [16]:
#print out the 46 sentences

n = 0
for sentence in processed_text2.sents:
    print(n, sentence)
    n+=1

0 Probably no other celebrity athlete will ever be in the same exact situation as O.J. Simpson – 70 years old asking for parole after a conviction for robbery, kidnapping, and conspiracy, some 23 years after one of the most sensationalized murder acquittals in American history.


1 But well over half of NBA and NFL players file for bankruptcy protection within 5 years of retirement.
2 Many will have diagnosed or undiagnosed brain injuries that prevent or hamper employment, and increase depression, mood swings, and bad judgement.
3 And bad judgment is what Simpson admitted as a cause for his botched attempt to reclaim intimate family photos and other memorabilia.
4 Many players may face a near-perfect storm in a similar setting – a desperate need for money and a misguided attempt to retrieve it, especially since the property involves his own name, image and likeness.


5 Some former athletes have also lived in a cocoon of celebrity, fighting the demons of being self-absorbed.
6 That can

In [17]:
#explore the various Parts of Speech and words used in the article

n = 0
for sentence in processed_text2.sents:
    for token in sentence:
        print(n, token, token.pos_, token.lemma_)
        n+=1

0 Probably ADV probably
1 no DET no
2 other ADJ other
3 celebrity NOUN celebrity
4 athlete NOUN athlete
5 will VERB will
6 ever ADV ever
7 be VERB be
8 in ADP in
9 the DET the
10 same ADJ same
11 exact ADJ exact
12 situation NOUN situation
13 as ADP as
14 O.J. PROPN o.j.
15 Simpson PROPN simpson
16 – PART –
17 70 NUM 70
18 years NOUN year
19 old ADJ old
20 asking VERB ask
21 for ADP for
22 parole NOUN parole
23 after ADP after
24 a DET a
25 conviction NOUN conviction
26 for ADP for
27 robbery NOUN robbery
28 , PUNCT ,
29 kidnapping NOUN kidnapping
30 , PUNCT ,
31 and CCONJ and
32 conspiracy NOUN conspiracy
33 , PUNCT ,
34 some DET some
35 23 NUM 23
36 years NOUN year
37 after ADP after
38 one NUM one
39 of ADP of
40 the DET the
41 most ADV most
42 sensationalized ADJ sensationalized
43 murder NOUN murder
44 acquittals NOUN acquittal
45 in ADP in
46 American ADJ american
47 history NOUN history
48 . PUNCT .
49 

 SPACE 


50 But CCONJ but
51 well INTJ well
52 over ADP over
53 half NOUN 

427 that ADJ that
428 led VERB lead
429 to ADP to
430 his ADJ -PRON-
431 conviction NOUN conviction
432 . PUNCT .
433 There ADV there
434 was VERB be
435 a DET a
436 conspiracy NOUN conspiracy
437 count NOUN count
438 , PUNCT ,
439 so CCONJ so
440 he PRON -PRON-
441 did VERB do
442 not ADV not
443 have VERB have
444 to PART to
445 hold VERB hold
446 a DET a
447 gun NOUN gun
448 to PART to
449 be VERB be
450 guilty ADJ guilty
451 . PUNCT .
452 Additionally ADV additionally
453 , PUNCT ,
454 Simpson PROPN simpson
455 did VERB do
456 not ADV not
457 testify VERB testify
458 during ADP during
459 the DET the
460 robbery NOUN robbery
461 trial NOUN trial
462 . PUNCT .
463 The DET the
464 jury NOUN jury
465 heard VERB hear
466 other ADJ other
467 testimony NOUN testimony
468 not ADV not
469 so ADV so
470 favorable ADJ favorable
471 . PUNCT .
472 Simpson NOUN simpson
473 and CCONJ and
474 players NOUN player
475 facing VERB face
476 a DET a
477 parole NOUN parole
478 hearing NOUN hearing
479 

In [18]:
#let's do a dependency parsing tree for the entire article

def pr_tree(word, level):
    if word.is_punct:
        return
    for child in word.lefts:
        pr_tree(child, level+1)
    print('\t'* level + word.text + ' - ' + word.dep_)
    for child in word.rights:
        pr_tree(child, level+1)

In [19]:
#print results of the parsing tree for the article

for sentence in processed_text2.sents:
    pr_tree(sentence.root, 0)
    print('=====================Next Tree=================')

	Probably - advmod
		no - det
		other - amod
		celebrity - compound
	athlete - nsubj
	will - aux
	ever - advmod
be - ROOT
	in - prep
			the - det
			same - amod
			exact - amod
		situation - pobj
=====================Next Tree=================
	But - cc
	well - nsubj
		over - prep
			half - pobj
				of - prep
						NBA - nmod
							and - cc
							NFL - conj
					players - pobj
file - ROOT
	for - prep
			bankruptcy - compound
		protection - pobj
	within - prep
			5 - nummod
		years - pobj
			of - prep
				retirement - pobj
=====================Next Tree=================
	Many - nsubj
	will - aux
	have - aux
diagnosed - ROOT
	or - cc
		undiagnosed - amod
		brain - compound
	injuries - conj
			that - nsubj
		prevent - relcl
			or - cc
			hamper - conj
				employment - dobj
				and - cc
					increase - compound
				depression - conj
					mood - compound
				swings - dobj
					and - cc
						bad - amod
					judgement - conj
=====================Next Tree=================
	And - cc
		bad -

**Observation:** This shows the various relationships throughout the article.  The fourth tree, references this sentence: "And bad judgment is what Simpson admitted as a cause for his botched attempt to reclaim intimate family photos and other memorabilia." 

It's interesting this tree pulls out the following subjects: Simpson and judgment.  These are then joined by the following verbs: admitted and the following modifiers: bad, botched.  Towards the end of the sentence it shows some sort of redemtion, where his admittance of "bad judgment" is a cause for his "botched" (modifies attempt) attempt (object) to reclaim (verb/action) his former fame ("photos and other memorabilia" - also other objects in the sentence).

**Note:** Let's take a closer look at this, just by taking out the first sentence and seeing the subject/object relationships.

In [20]:
subtext = """Probably no other celebrity athlete will ever be in the same exact situation as O.J. Simpson – 70 years old asking for parole after a conviction for robbery, kidnapping, and conspiracy, some 23 years after one of the most sensationalized murder acquittals in American history."""

In [21]:
processed_subtext = processor(subtext)
processed_subtext

Probably no other celebrity athlete will ever be in the same exact situation as O.J. Simpson – 70 years old asking for parole after a conviction for robbery, kidnapping, and conspiracy, some 23 years after one of the most sensationalized murder acquittals in American history.

In [22]:
for sentence in processed_subtext.sents:
    pr_tree(sentence.root, 0)
    print('=====================End Tree=================')

	Probably - advmod
		no - det
		other - amod
		celebrity - compound
	athlete - nsubj
	will - aux
	ever - advmod
be - ROOT
	in - prep
			the - det
			same - amod
			exact - amod
		situation - pobj
=====================End Tree=================


**Observation:** It appears that there aren't many objects for the subject. It appears that the situation could apply to other "celebrity athletes". 

### 4. Show the most common entities and their types

In [23]:
#explore the various entities and their labeled types in the article

for entity in processed_text2.ents:
    print(entity, entity.label_)
    

O.J. Simpson – PERSON
70 years old DATE
some 23 years DATE
one CARDINAL
American NORP
well over half CARDINAL
NBA ORG
NFL ORG
within 5 years DATE
Simpson PERSON
Simpson PERSON
19 DATE
Simpson PERSON
Simpson’s PERSON
Simpson PERSON
Simpson PERSON
Simpson PERSON
Simpson PERSON
Board ORG
Simpson PERSON
Simpson PERSON
1 CARDINAL
9.5 CARDINAL
33 CARDINAL
CNN ORG
Mark Geragos PERSON
one-year DATE
2 CARDINAL
3 CARDINAL
one CARDINAL
Simpson PERSON
911 CARDINAL
Simpson’s PERSON
the year DATE
one CARDINAL
Simpson PERSON
Simpson PERSON
October, 2017 DATE
Simpson PERSON


In [24]:
#from the above list, it appears that Simpson is the most common entity (type = PERSON)
#let's create a dictionary and store the various PERSON entities in it

persons = []

for entity in processed_text2.ents:
    if entity.label_ == 'PERSON':
        persons.append(entity)


        

In [25]:
# to show the most frequent PERSON entities, let's use the Counter

persons_frequency = Counter(persons)
common_persons = persons_frequency.most_common(5)
common_persons

[(O.J. Simpson –, 1), (Simpson, 1), (Simpson, 1), (Simpson, 1), (Simpson’s, 1)]

**Observation:** As suspected, Simpson (or OJ Simpson) is in the top 5 of frequent entities.  

### 5. Find entities and their dependency (hint: entity.root.head)

**Note:** From the question above, it appears there are very few entities (a couple of PERSON, a few CARDINALs, a couple of DATEs, and a couple ORGs).  Probably best to explore the PERSON and ORG entity types for this question.


In [122]:
for entity in processed_text2.ents:
    print(entity.label_, entity, entity.root.head)

PERSON O.J. Simpson – situation
DATE 70 years old –
DATE some 23 years robbery
CARDINAL one after
NORP American history
CARDINAL well over half file
ORG NBA players
ORG NFL NBA
DATE within 5 years file
PERSON Simpson admitted
PERSON Simpson to
DATE 19 was
PERSON Simpson mistake
PERSON Simpson’s ,
PERSON Simpson admitted
PERSON Simpson risked
PERSON Simpson testify
PERSON Simpson remember
ORG Board looking
PERSON Simpson was
PERSON Simpson had
CARDINAL 1 of
CARDINAL 9.5 33
CARDINAL 33 years
ORG CNN commentator
PERSON Mark Geragos said
DATE one-year sentence
CARDINAL 2 had
CARDINAL 3 being
CARDINAL one exists
PERSON Simpson had
CARDINAL 911 calls
PERSON Simpson’s statement
DATE the year of
CARDINAL one member
PERSON Simpson receive
PERSON Simpson faced
DATE October, 2017 in
PERSON Simpson benefits


**Observation:** PERSON appears to be the most common entity type, with Simpson being the most frequent entity. The dependencies related to Simpson are interesting and include (verbs, adverbs): "admitted", "risked", "testify", "situation", "mistake", "benefits".

### 6. Find the most similar noun chunks in the article.

**Note:** Best to use word vectorization to compare similarity among noun_chunks. Let's compare similarity to a select few of the noun chunks, which are added to our comparison vocabulary list.

In [63]:
#first append all the noun chunks into an empty list, in order to then run the similarity scores
n=0
nounlist = []
for noun_chunk in processed_text2.noun_chunks:
    nounlist.append(noun_chunk)
    print(n, noun_chunk)
    n+=1

0 no other celebrity athlete
1 the same exact situation
2 O.J. Simpson
3 parole
4 a conviction
5 robbery
6 kidnapping
7 conspiracy
8 the most sensationalized murder acquittals
9 American history
10 half
11 NBA and NFL players
12 bankruptcy protection
13 5 years
14 retirement
15 undiagnosed brain injuries
16 employment
17 mood swings
18 bad judgement
19 bad judgment
20 what
21 Simpson
22 a cause
23 his botched attempt
24 intimate family photos
25 other memorabilia
26 Many players
27 a near-perfect storm
28 setting
29 a desperate need
30 money
31 a misguided attempt
32 it
33 the property
34 his own name
35 image
36 likeness
37 Some former athletes
38 a cocoon
39 celebrity
40 the demons
41 a failure
42 the role
43 contrition
44 remorse
45 Those character attributes
46 a judge
47 jury
48 a parole board
49 forgiveness
50 We
51 crimes
52 former athletes
53 fame
54 greater number
55 we
56 a version
57 these circumstances
58 lessons
59 therefrom
60 I
61 the bad guy
62 somebody
63 the more stra

In [75]:
#create a vocabuilary list, the list of words we're comparing our word in the noun chunk list to
#based on the example in the lecture notes e.g., fruit = proc_fruits.vocab['fruit']

athletes = processed_text2.vocab['athletes'] #noun chunk 67
team = processed_text2.vocab['team'] #noun chunk 85
thugs = processed_text2.vocab['thugs'] #noun chunk 92
futureplayers = processed_text2.vocab['Future players'] #noun chunk 109
stardom = processed_text2.vocab['stardom'] #noun chunk 159

In [97]:
#now take the noun chunk list and iterate its similarity scores through the list of vocabulary words.  
#save the results into a list (similaritylist)

n=0
similaritylist = []
for word in nounlist:
    similaritylist.append(word)
    similaritylist.append(word.similarity(athletes))
    similaritylist.append(word.similarity(team))
    similaritylist.append(word.similarity(thugs))
    similaritylist.append(word.similarity(futureplayers))
    similaritylist.append(word.similarity(stardom))

    n+=1

similaritylist

[no other celebrity athlete,
 0.63460167269968704,
 0.43964598226565765,
 0.0,
 0.0,
 0.0,
 the same exact situation,
 0.26005855782927617,
 0.39908922973354993,
 0.0,
 0.0,
 0.0,
 O.J. Simpson,
 0.0,
 0.0,
 0.0,
 0.0,
 0.0,
 parole,
 0.0,
 0.0,
 0.0,
 0.0,
 0.0,
 a conviction,
 0.096884066595711554,
 0.33833234293954723,
 0.0,
 0.0,
 0.0,
 robbery,
 0.0,
 0.0,
 0.0,
 0.0,
 0.0,
 kidnapping,
 0.0,
 0.0,
 0.0,
 0.0,
 0.0,
 conspiracy,
 0.12275444752398373,
 0.12070107642446223,
 0.0,
 0.0,
 0.0,
 the most sensationalized murder acquittals,
 0.30398245435977134,
 0.35892138335405505,
 0.0,
 0.0,
 0.0,
 American history,
 0.31223094244992072,
 0.30135227117725405,
 0.0,
 0.0,
 0.0,
 half,
 0.24283032476039185,
 0.38734450914469537,
 0.0,
 0.0,
 0.0,
 NBA and NFL players,
 0.48133309901318788,
 0.57648677877820809,
 0.0,
 0.0,
 0.0,
 bankruptcy protection,
 0.1611141515294072,
 0.1682029145473378,
 0.0,
 0.0,
 0.0,
 5 years,
 0.21154810160565918,
 0.32453259565075177,
 0.0,
 0.0,
 0.0,
 re

**Note:** To make this list easier to read, I attempted to create a dataframe from these results, but kept getting errors (I attempted pd.DataFrame, np.array)...not to great success.  

In [89]:
import pandas as pd
import numpy as np

columnlabels = ['Noun Chunk', 'Similarity to Athlete', 'Similarity to Team', 'Similarity to Thugs', 'Similarity to Future Players', 'Similarity to Stardom']
dataframe = pd.DataFrame(similaritylist, columns=columnlabels)

dataframe


TypeError: 'numpy.float64' object is not iterable

**Observation:** Depending on the noun chunk pairings, here are a few of the top results:

*Top 4 Noun chunks most similar to "Athlete"*
1. former athletes (0.82) 
2. Some former athletes (0.77) 
3. No Other Celebrity Athlete (0.63)
4. Many players (0.54)

*Top 4 Noun chunks most similar to "Team"*
1. NBA and NFL players (0.57)
2. Some former athletes (0.54)
3. Many players (0.54)
4. former athletes (0.52)

For the remainder of the comparision words, "thugs", "Future players", and "stardom", the similiarity scores were 0.0.  This indicates that perhaps there was too little to compare (uniqueness of the word) or maybe some depreciating issue with the model (which doesn't make too much sense as comparisions to "athlete" and "team", yielded many and varied results.




